src/12_RefactoringGolf/hole1/kata.ts

```diff
@@ -1,107 +1,142 @@
   }
 
   public Winner(): string {
-    //if the positions in first row are taken
-    if (
-      this._toto.TileAt(0, 0)!.Symbol != ' ' &&
-      this._toto.TileAt(0, 1)!.Symbol != ' ' &&
-      this._toto.TileAt(0, 2)!.Symbol != ' '
-    ) {
-      //if first row is full with same symbol
-      if (
-        this._toto.TileAt(0, 0)!.Symbol == this._toto.TileAt(0, 1)!.Symbol &&
-        this._toto.TileAt(0, 2)!.Symbol == this._toto.TileAt(0, 1)!.Symbol
-      ) {
-        return this._toto.TileAt(0, 0)!.Symbol;
-      }
-    }
+    return this._board.findRowFullWithSamePlayer();
+  }
+}
 
-    //if the positions in first row are taken
-    if (
-      this._toto.TileAt(1, 0)!.Symbol != ' ' &&
-      this._toto.TileAt(1, 1)!.Symbol != ' ' &&
-      this._toto.TileAt(1, 2)!.Symbol != ' '
-    ) {
-      //if middle row is full with same symbol
-      if (
-        this._toto.TileAt(1, 0)!.Symbol == this._toto.TileAt(1, 1)!.Symbol &&
-        this._toto.TileAt(1, 2)!.Symbol == this._toto.TileAt(1, 1)!.Symbol
-      ) {
-        return this._toto.TileAt(1, 0)!.Symbol;
-      }
-    }
+class Tile {
+  private x: number = 0;
+  private y: number = 0;
+  private symbol: string = ' ';
 
-    //if the positions in first row are taken
-    if (
-      this._toto.TileAt(2, 0)!.Symbol != ' ' &&
-      this._toto.TileAt(2, 1)!.Symbol != ' ' &&
-      this._toto.TileAt(2, 2)!.Symbol != ' '
-    ) {
-      //if middle row is full with same symbol
-      if (
-        this._toto.TileAt(2, 0)!.Symbol == this._toto.TileAt(2, 1)!.Symbol &&
-        this._toto.TileAt(2, 2)!.Symbol == this._toto.TileAt(2, 1)!.Symbol
-      ) {
-        return this._toto.TileAt(2, 0)!.Symbol;
-      }
-    }
+  constructor(x: number, y: number, symbol: string) {
+    this.x = x;
+    this.y = y;
+    this.symbol = symbol;
+  }
 
-    return ' ';
+  get Symbol() {
+    return this.symbol;
+  }
+
+  get isNotEmpty() {
+    return this.Symbol !== emptyPlay;
+  }
+
+  hasSameSymbolAs(other: Tile) {
+    return this.Symbol === other.Symbol;
+  }
+
+  hasSameCoordinatesAs(other: Tile) {
+    return this.x == other.x && this.y == other.y;
   }
-}
 
-interface Tile {
-  X: number;
-  Y: number;
-  Symbol: string;
+  updateSymbol(newSymbol: string) {
+    this.symbol = newSymbol;
+  }
 }
 
 class Board {
   private _plays: Tile[] = [];
 
   constructor() {
-    for (let i = 0; i < 3; i++) {
-      for (let j = 0; j < 3; j++) {
-        const tile: Tile = { X: i, Y: j, Symbol: ' ' };
-        this._plays.push(tile);
+    for (let x = firstRow; x <= thirdRow; x++) {
+      for (let y = firstColumn; y <= thirdColumn; y++) {
+        this._plays.push(new Tile(x, y, emptyPlay));
       }
     }
   }
 
   public TileAt(x: number, y: number): Tile {
-    return this._plays.find((t: Tile) => t.X == x && t.Y == y)!;
+    return this._plays.find((t: Tile) => t.hasSameCoordinatesAs(new Tile(x, y, emptyPlay)))!;
   }
 
   public AddTileAt(symbol: string, x: number, y: number): void {
```

Comment:
Rename so it dosent start with capital letter since it is a method same for TileAt